bin/dax_tools/GenerateSpiderTemplate

```diff
@@ -0,0 +1,339 @@
+    your arguments:
+        ...
+
+    :return: argument parser object created by parse_args()
+    '''
+    from argparse import ArgumentParser
+    ap = spiders.get_scan_argparser("{name}", "{purpose}")
+
+    #
+    # ADD YOUR OTHER OPTIONS FOR THIS SPECIFIC SPIDER
+    # example to add one options:
+    # ap.add_argument("--option", dest="option_id", default=None, help="Option description.", required=True)
+    #
+
+    return ap.parse_args()
+
+class Spider_{name}(ScanSpider):
+    '''
+        Scan Spider class to do: {purpose}
+
+        :param spider_path: spider file path
+        :param jobdir: directory for temporary files
+        :param xnat_project: project ID on XNAT
+        :param xnat_subject: subject label on XNAT
+        :param xnat_session: experiment label on XNAT
+        :param xnat_scan: scan label on XNAT
+        :param xnat_host: host for XNAT if not set in environment variables
+        :param xnat_user: user for XNAT if not set in environment variables
+        :param xnat_pass: password for XNAT if not set in environment variables
+        :param suffix: suffix to the assessor creation
+    '''
+    def __init__(self, spider_path, jobdir, xnat_project, xnat_subject, xnat_session, xnat_scan,
+                 xnat_host=None, xnat_user=None, xnat_pass=None, suffix=""):
+        super(Spider_{name}, self).__init__(spider_path, jobdir, xnat_project, xnat_subject, xnat_session, xnat_scan,
+                                            xnat_host, xnat_user, xnat_pass, suffix)
+        self.inputs = list()
+
+    def pre_run(self, argument_parse):
+        '''
+            Method to download data from XNAT
+
+            :param argument_parse: argument parser object return by parse_args()
+        '''
+        resource = 'XXXX' #resource to download from the scan on XNAT
+        folder = os.path.join(self.jobdir, 'inputs')
+        os.makedirs(folder)
+        self.inputs.extend(self.download(self.xnat_scan, resource, folder))
+
+        #
+        # YOU CAN ADD MORE line to download other data LIKE
+        #    self.inputs.extend(self.download(**kwargs))
+        #
+
+    def run(self):
+        '''
+            Method running the process for the spider on the inputs data
+        '''
+        #
+        # CODE THAT YOU WANT TO RUN ON THE DATA
+        #
+
+    def finish(self):
+        '''
+            Method to copy the results in the Spider Results folder dax.RESULTS_DIR
+        '''
+        self.time_writer('Results saved in folder: %s' % (RESULTS_DIR))
+        results_dict = {{'PDF': pdfpath,
+                        #
+                        # ADD OTHER RESULTS YOU WANT TO SAVE
+                        #
+                        }}
+        self.upload_dict(results_dict)
+        self.end()
+
+if __name__ == '__main__':
+    args = parse_args()
+    # generate spider object:
+    spider_obj = Spider_{name}(spider_path=sys.argv[0],
+                               xnat_project=args.proj_label,
```

Comment:
ok, I think I fixed both issues.

Thanks for reporting them Brian.

Let me know if you anymore of them.

Cheers,

Ben

On Wed, Sep 30, 2015 at 7:48 PM, Brian D. Boyd <notifications@github.com>
wrote:

> In bin/dax_tools/GenerateSpiderTemplate
> <https://github.com/VUIIS/dax/pull/21#discussion_r40834691>:
>
> > +            Method to copy the results in the Spider Results folder dax.RESULTS_DIR
> > +        '''
> > +        self.time_writer('Results saved in folder: %s' % (RESULTS_DIR))
> > +        results_dict = {{'PDF': pdfpath,
> > +                        #
> > +                        # ADD OTHER RESULTS YOU WANT TO SAVE
> > +                        #
> > +                        }}
> > +        self.upload_dict(results_dict)
> > +        self.end()
> > +
> > +if __name__ == '__main__':
> > +    args = parse_args()
> > +    # generate spider object:
> > +    spider_obj = Spider_{name}(spider_path=sys.argv[0],
> > +                               xnat_project=args.proj_label,
>
> we are missing the jobdir argument here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/VUIIS/dax/pull/21/files#r40834691>.
>